src/sentry/static/sentry/app/components/organizations/globalSelectionHeader/index.jsx

```diff
@@ -301,7 +301,8 @@ class GlobalSelectionHeader extends React.Component {
   handleUpdateProjects = () => {
     const {projects} = this.state;
     updateProjects(projects, this.getRouter(), this.getUpdateOptions());
-    this.setState({projects: null});
+    updateEnvironments(null, this.getRouter(), this.getUpdateOptions());
+    this.setState({projects: null, environments: null});
```

Comment:
Didn't we have code that looked for intersections between the currently selected environments and the environments belonging to the newly selected projects?

Reply:
Ah we do, it just doesn't completely work

Reply:
Ok. Should we remove it? It lives https://github.com/getsentry/sentry/blob/26939bf07cae41820d8d793e9a588b09f6bc117e/src/sentry/static/sentry/app/components/organizations/multipleEnvironmentSelector.jsx#L62-L78